Sources/BigInt/BigInt.swift

```diff
@@ -0,0 +1,816 @@
+      rtempHigh += 1
+    }
+
+    (rtempLow, overflow) = rtempLow.addingReportingOverflow(nextUdigit)
+    if overflow {
+      rtempHigh += 1
+    }
+
+    while true {
+      if (tempHigh > rtempHigh) || ((tempHigh == rtempHigh) && (tempLow > rtempLow)) {
+        qhat -= 1
+        (rhat, overflow) = rhat.addingReportingOverflow(divisor)
+        if !overflow {
+          continue
+        } else {
+          break
+        }
+      } else {
+        break
+      }
+    }
+
+    return qhat
+  }
+
+  @usableFromInline
+  internal static func _div(lhs: BigInt, rhs: BigInt) -> (quotient: BigInt, remainder: BigInt) {
+    precondition(rhs != _digits[0], "Division by zero error!")
+
+    if lhs.words.count == 1, rhs.words.count == 1 {
+      let (quot, rem) = Int(bitPattern: lhs.words[0]).quotientAndRemainder(dividingBy: Int(bitPattern: rhs.words[0]))
+      return (BigInt(words: [UInt(bitPattern: quot)]), BigInt(words: [UInt(bitPattern: rem)]))
+    }
+
+    let lhsIsNeg = lhs._isNegative
+    let rhsIsNeg = rhs._isNegative
+
+    var lhsWords = lhsIsNeg ? (-lhs).words : lhs.words
+    var rhsWords = rhsIsNeg ? (-rhs).words : rhs.words
+
+    while rhsWords[rhsWords.endIndex - 1] == 0, rhsWords.count > 2 {
+      rhsWords.removeLast()
+    }
+
+    if rhsWords.count > lhsWords.count { return (0, lhs) }
+
+    if rhsWords.count == 1 {
+      rhsWords.append(0)
+    }
+
+    func normalize(x: UInt) -> UInt {
+      #if arch(arm64) || arch(x86_64)
+        if x == 0 {
+          return 64
+        }
+        var x = UInt64(x)
+        var n: UInt = 0
+        if x <= 0x0000_0000_FFFF_FFFF {
+          n += 32
+          x <<= 32
+        }
+        if x <= 0x0000_FFFF_FFFF_FFFF {
+          n += 16
+          x <<= 16
+        }
+        if x <= 0x00FF_FFFF_FFFF_FFFF {
+          n += 8
+          x <<= 8
+        }
+        if x <= 0x0FFF_FFFF_FFFF_FFFF {
+          n += 4
+          x <<= 4
+        }
+        if x <= 0x3FFF_FFFF_FFFF_FFFF {
+          n += 2
+          x <<= 2
+        }
+        if x <= 0x7FFF_FFFF_FFFF_FFFF {
+          n += 1
+        }
+
+        return n
+      #else
+        if x == 0 {
+          return 32
+        }
+
+        var x = x
+        var n: UInt = 0
+        if x <= 0x0000_FFFF {
+          n += 16
+          x <<= 16
+        }
+        if x <= 0x00FF_FFFF {
+          n += 8
+          x <<= 8
+        }
+        if x <= 0x0FFF_FFFF {
+          n += 4
+          x <<= 4
+        }
+        if x <= 0x3FFF_FFFF {
+          n += 2
+          x <<= 2
+        }
+        if x <= 0x7FFF_FFFF {
+          n += 1
+        }
+
+        return n
+      #endif
+    }
+
+    if lhsWords.count < rhsWords.count {
+      for _ in 0 ..< (rhsWords.count - lhsWords.count) {
+        lhsWords.append(0)
+      }
+    }
+
+    let m = lhsWords.count
+    let n = rhsWords.count
+
+    let bitWidth = UInt(UInt.bitWidth)
+
+    let s = normalize(x: rhsWords[n - 1])
+    let rn = UnsafeMutablePointer<UInt>.allocate(capacity: n)
```

Comment:
Allocating unsafe pointers here seems unnecessary -- wouldn't a temporary array work here?

Reply:
In my tests, allocating the memory this way was _significantly_ faster than a temporary array.